src/charm.py

```diff
@@ -197,14 +227,19 @@ def _get_s3_env(self):
 
         return s3_env
 
-    def _create_discourse_environment_settings(self):
-        """Create the pod environment config from the existing config."""
+    def _create_discourse_environment_settings(self) -> Dict:
+        """Create a layer config based on our current configuration.
+
+        Returns:
+            Dictionary with all the environment settings.
+        """
         # Get redis connection information from the relation.
         redis_hostname = None
         redis_port = 6379
-        for redis_unit in self._stored.redis_relation:
-            redis_hostname = self._stored.redis_relation[redis_unit]["hostname"]
-            redis_port = self._stored.redis_relation[redis_unit]["port"]
+        # This is the current recommended way of accessing the relation data.
+        for redis_unit in self._stored.redis_relation:  # type: ignore
+            redis_hostname = self._stored.redis_relation[redis_unit].get("hostname")  # type: ignore
+            redis_port = self._stored.redis_relation[redis_unit].get("port")  # type: ignore
             logger.debug(
                 "Got redis connection details from relation of %s:%s", redis_hostname, redis_port
             )
```

Comment:
I know this wasn't changed in this PR, it seems that the configuration uses the `redis_hostname` and `redis_port` for the last `redis_unit` in `self._stored.redis_relation` and does so implicitly by relying on changes to a variable in a loop, would it be worth reviewing that logic to see if it is still appropriate? It could be more explicit. Maybe better for another PR?